libpod/boltdb_state.go

```diff
@@ -783,6 +783,50 @@ func (s *BoltState) AllContainers() ([]*Container, error) {
 	return ctrs, nil
 }
 
+// RewriteContainerConfig rewrites a container's configuration.
+// WARNING: This function is DANGEROUS. Do not use without reading the full
+// comment on this function in state.go.
+func (s *BoltState) RewriteContainerConfig(ctr *Container, newCfg *ContainerConfig) error {
+	if !s.valid {
+		return ErrDBClosed
```

Comment:
I wonder why `ErrDBClosed` is used. Is the deduction invalid -> database is closed always true? I mostly wonder because of `defer s.closeDBCon(db)` in line 807.

Reply:
Valid is... less meaningful than it used to be, in bolt? We initially maintained a single connection, which we reused, and then valid made sense (if valid == false, the connection was closed). Now, we make a new connection for each function, which means valid can be true, and no connection is open... In the current implementation, valid == false mostly means we've "shut down" the database, and are refusing to make new connections

Reply:
Thanks for insights!